cot/src/openapi.rs

```diff
@@ -0,0 +1,1179 @@
+//! OpenAPI integration for Cot.
+//!
+//! This module provides traits and utilities for generating OpenAPI
+//! documentation for Cot applications. The idea is to be able to use Cot's
+//! existing request handlers and extractors to generate OpenAPI documentation
+//! automatically.
+//!
+//! # Usage
+//!
+//! 1. Add [`#[derive(schemars::JsonSchema)]`](schemars::JsonSchema) to the
+//!    types used in the extractors and response types.
+//! 2. Use [`ApiMethodRouter`](crate::router::method::openapi::ApiMethodRouter)
+//!    to set up your API routes and register them with a router.
+//! 3. Register your
+//!    [`ApiMethodRouter`](crate::router::method::openapi::ApiMethodRouter)s
+//!    with a [`Router`](crate::router::Router) using
+//!    [`Route::with_api_handler`](crate::router::Route::with_api_handler).
+//! 4. Register the [`SwaggerUi`](crate::openapi::swagger_ui::SwaggerUi) app
+//!    inside [`Project::register_apps`](crate::project::Project::register_apps).
```

Comment:
A few notes:

1. Mention `Route::with_api_handler_name`
2. We could mention the convenience functions in 2. (eg. `api_get`, `api_post`, etc.)
3. I think it might be worth mentioning the need to set up the `/swagger` route as well (and `StaticFilesMiddleware`)